handler.py

```diff
@@ -1,280 +1,70 @@
-        separator = "<br />" if len(eval_function_feedback[0]) > 0 else ""
-        matched_case.update(
-            {
-                "feedback": matched_case.get("feedback", "")
-                + separator
-                + eval_function_feedback[0]
-            }
-        )
+    elif command in ("eval", "grade"):
+        response = commands.evaluate(event)
+        validator = ResBodyValidators.EVALUATION
 
-    if len(matches) == 1:
-        # warnings += [{"case": matches[0]}]
-        return matched_case, warnings
+    elif command == "preview":
+        response = commands.preview(event)
+        validator = ResBodyValidators.PREVIEW
 
+    elif command == "healthcheck":
+        response = commands.healthcheck()
+        validator = ResBodyValidators.HEALTHCHECK
     else:
-        s = ", ".join([str(m) for m in matches])
-        warnings += [
-            {
-                "message": f"Cases {s} were matched. Only the first one's feedback was returned"
-            }
-        ]
-        return matched_case, warnings
+        response = Response(
+            error=ErrorResponse(message=f"Unknown command '{command}'.")
+        )
+        validator = ResBodyValidators.GENERIC
 
+    validate.body(response, validator)
 
-"""
-    Main Handler Function
-"""
+    return response
 
 
-def handler(event, context={}):
+def handler(event: JsonType, context: JsonType = {}) -> HandlerResponse:
     """
     Main function invoked by AWS Lambda to handle incoming requests.
     ---
-    This function invokes the handler function for that particular command and returns
-    the result. It also performs validation on the response body to make sure it follows
+    This function invokes the handler function for that particular command
+    and returns
+    the result. It also performs validation on the response body to make sure
+    it follows
     the schema set out in the request-response-schema repo.
     """
     headers = event.get("headers", dict())
     command = headers.get("command", "eval")
 
-    if command == "healthcheck":
-        response = handle_healthcheck_command()
-        response_error = v.validate_healthcheck_response(response)
-
-    # Remove once all funcs update to V2
-    elif command == "eval" or command == "grade":
-        response = handle_eval_command(event)
-        response_error = v.validate_eval_response(response)
-
-    elif command == "preview":
-        response = handle_preview_command(event)
-        response_error = v.validate_preview_response(response)
+    try:
+        return handle_command(event, command)
 
-    elif command == "docs-dev" or command == "docs":
-        return docs.send_dev_docs()
+    except ParseError as e:
+        error = ErrorResponse(message=e.message, detail=e.error_thrown)
 
-    elif command == "docs-user":
-        return docs.send_user_docs()
+    except ValidationError as e:
+        error = ErrorResponse(message=e.message, detail=e.error_thrown)
```

Comment:
Can collapse this block with the one above:

```suggestion
    except (ParseError, ValidationError) as e:
        error = ErrorResponse(message=e.message, detail=e.error_thrown)
```

Reply:
Agreed - will commit this.